File: translation_output/wizardcoder/codenet/java/python-sanitized/p03146.py
import java.util.HashSet
import java.util.Scanner
import java.util.Set

class p03146:
    sc = Scanner(System.in)

    def main(self):
        s = sc.nextInt()
        set = HashSet()
        set.add(s)
        i = 1
        while True:
            i += 1
            if s % 2 == 0:
                s = s // 2
            else:
                s = 3 * s + 1
            if set.contains(s):
                print(i)
                return
            else:
                set.add(s)

p03146().main()

File: dataset/Translation/wizardcoder/codenet_python_java/p03146/p03146.java
import java.util.*;

public class p03146 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int s = sc.nextInt();
        solve(s);
    }

    public static void solve(int s) {
        int m = 1;
        int pre = s;
        Set<Integer> st = new HashSet<>();
        st.add(s);
        while (true) {
            m++;
            if (pre % 2 == 0) {
                pre /= 2;
            } else {
                pre = 3 * pre + 1;
            }
            if (st.contains(pre)) {
                System.out.println(m);
                return;
            }
            st.add(pre);
        }
    }
}